src/nvim/mouse.c

```diff
@@ -328,6 +328,43 @@ static int do_popup(int which_button, int m_pos_flag, pos_T m_pos)
   return jump_flags;
 }
 
+/// Handle scrollbar click or drag by calculating the target file position
+/// based on mouse position and scrolling to that location.
+static void mouse_handle_scrollbar(win_T *wp)
+{
+  int height = wp->w_view_height;
+  linenr_T line_count = wp->w_buffer->b_ml.ml_line_count;
+  linenr_T end_lnum = line_count;
+  int64_t end_vcol = -1;
+  int total_lines = wp->w_buffer->b_ml.ml_vline_count;
+
+  if (total_lines > height) {
+    int thumb_size = MAX(1, (height * height) / total_lines);
+    int scrollable_area = height - thumb_size;
+    int click_row = mouse_row - (wp->w_winrow + wp->w_border_adj[0]);
+    click_row = MAX(0, MIN(click_row, scrollable_area));
+    int target_screen_offset = (click_row * (total_lines - height)) / scrollable_area;
+    linenr_T target_topline = 1;
+    int accumulated = 0;
+
+    while (target_topline < line_count && accumulated < target_screen_offset) {
+      end_lnum = target_topline;
+      end_vcol = -1;
+      int line_height = (int)win_text_height(wp, target_topline, -1, &end_lnum, &end_vcol, NULL, 1);
```

Comment:
```suggestion
      int line_height = plines_win(wp, target_topline, false);
```

Reply:
If replace `win_text_height` with `plines_win`, the loop would incorrectly count folded lines. For example, if lines 10-50 are folded (41 lines), the loop would add 41 screen lines instead of 1.  so need a manualy fold handling  like this ...🤔

```c
while (target_topline < line_count && accumulated < target_screen_offset) {
  linenr_T fold_end = target_topline;
  if (hasFolding(wp, target_topline, NULL, &fold_end)) {
    accumulated += 1;
    target_topline = fold_end + 1;
  } else {
    accumulated += plines_win(wp, target_topline, false);
    target_topline++;
  }
}
````

Reply:
Oh right, I forgot that `end_lnum` is set to the line number after the fold.

Reply:
Hmm, does `win_text_height` work for this case? If `start_lnum == *end_lnum`, `end_lnum` is set to the start of the fold, not the end.